pcl_ros/CMakeLists.txt

```diff
@@ -118,8 +119,19 @@ ament_target_dependencies(pcl_ros_tf
 #
 ### Tools
 #
-#add_executable(pcd_to_pointcloud tools/pcd_to_pointcloud.cpp)
-#target_link_libraries(pcd_to_pointcloud ${Boost_LIBRARIES} ${catkin_LIBRARIES} ${PCL_LIBRARIES})
+add_library(pcd_to_pointcloud_lib SHARED tools/pcd_to_pointcloud.cpp)
+target_link_libraries(pcd_to_pointcloud_lib
+  ${PCL_LIBRARIES})
+target_include_directories(pcd_to_pointcloud_lib PUBLIC
+  ${PCL_INCLUDE_DIRS})
+ament_target_dependencies(pcd_to_pointcloud_lib
+  rclcpp
+  rclcpp_components
+  sensor_msgs
+  pcl_conversions)
+rclcpp_components_register_node(pcd_to_pointcloud_lib
+  PLUGIN "pcl_ros::PCDGenerator"
```

Comment:
nitpick: I would make the name of the plugin more closely resemble the name of the executable. I don't think `PCDGenerator` is an accurate description for what it does.

Reply:
I picked the old name used in ROS 1.
If we want to update the name, `PCDPublisher` may be a good one.

@SteveMacenski what do you think?

Reply:
Yeah, more in line with the file name would be good

Reply:
Renamed in https://github.com/ros-perception/perception_pcl/pull/359/commits/467ba51b03456b64f39dd8c50317965fafb8f2e2.